File: 프로그래머스/lv1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string phone_number) {
    
    int cnt = 1;
    for(int i = phone_number.size() - 1; i >= 0; i--)
    {
        if(cnt > 4) phone_number[i] = '*';
        ++cnt;
    }
    
    return phone_number;
}

File: 프로그래머스/1/12948. 핸드폰 번호 가리기/핸드폰 번호 가리기.java
class Solution {
    public String solution(String phone_number) {
        int cnt = 0;
        
        String returnPN = "";
        
        for(int i = phone_number.length() -1; i >= 0; i--) {
            if(cnt == 4) {
                returnPN = "*" + returnPN;
            } else {
                returnPN = "" + phone_number.charAt(i) + returnPN;
                ++cnt;
            }
        }
        return returnPN;
    }
}